pkg/runtime/reconciler.go

```diff
@@ -207,13 +207,6 @@ func (r *reconciler) sync(
 			"diff", diffReporter.String(),
 			"arn", latest.Identifiers().ARN(),
 		)
-		// Before we update the backend AWS service resources, let's first update
-		// the latest status of CR which was retrieved by ReadOne call.
-		// Else, latest read status is lost in-case Update call fails with error.
-		err = r.patchResource(ctx, desired, latest)
-		if err != nil {
-			return err
-		}
```

Comment:
This was specifically added by @vijtrip2 to handle scenarios in APIGateway v2. Without a test showing whether removing this would impact other service controllers, I'm hesitant to merge this, even though I believe you are correct in removing it here.

Vijay, could you please advise?

Reply:
taking a look

Reply:
This logic was added by @kumargauravsharma and it will not impact apigatewayv2 changes that I made in readOne operation.

I will add the update operation in apigatewayv2-e2e test as part of issue #539 

I am fine with merging this change in, but if @kumargauravsharma can wait until I finish #539, that would be great.
I should be able to finish it by 12/1/2020 End of Day PST.

Reply:
PR for adding apigatewayv2 update-api e2e test. https://github.com/aws/aws-controllers-k8s/pull/544